rest_service.go

```diff
@@ -83,26 +86,38 @@ func (s *Service) App() *gimlet.APIApp {
 	app.AddRoute("/clear").Version(1).Post().Handler(s.clearManager)
 	app.AddRoute("/close").Version(1).Delete().Handler(s.closeManager)
 
-	go s.backgroundPrune()
+	go s.backgroundPrune(ctx)
 
 	return app
 }
 
-func (s *Service) backgroundPrune() {
+func (s *Service) backgroundPrune(ctx context.Context) {
+	defer func() {
+		err := recovery.HandlePanicWithError(recover(), nil, "background pruning")
+		if ctx.Err() != nil || err == nil {
```

Comment:
Why return if the error is nil?

Reply:
because it's in a defer: if the context was canceled *or* if there was no panic (and we're just returning,) then we want to exit, otherwise respawn the thread